newIDE/app/src/EventsSheet/InstructionEditor/NewInstructionEditorDialog.js

```diff
@@ -243,7 +243,7 @@ export default function NewInstructionEditorDialog({
     <ResponsiveWindowMeasurer>
       {windowWidth => (
         <Dialog
-          onApply={onSubmit}
+          onApply={instructionType && onSubmit}
```

Comment:
instructType ? onSubmit : null

And adapt the onApply type to be `onApply?: ?() => void`